posts/2020/12/jupyterlab-winter-theme.md

```diff
@@ -0,0 +1,347 @@
+If you want to change or add icons and keep them matching, finding one from this system will fit best. The [Material Design system](https://material.io/design/iconography/system-icons.html) 
+also points out some of their icon design principles which are good to follow if you need to make custom icons that 
+match with the rest. Use SVGs, not PNGs, and remember to give it a [tooltip](https://en.wikipedia.org/wiki/Tooltip). 
+Most of all, make sure to give it an ARIA label (like [this recommendation](https://gomakethings.com/icon-accessibility-and-aria-label/)).
+
+# Modifying the Theme Variables
+
+After each modification the watch process will build the extension for you. 
+No need to stop and restart JupyterLab server; simply refresh the page.
+
+Now we are going to modify some values in the file `variables.css` in our
+project. This file controls many of the colors of JupyterLab, and is a nice place
+to start to change the overall color scheme before doing more detailed
+customisation.
+
+We'll try to update the current theme from orange and blue to more blue-ish tones, 
+which tend to remind me of the holiday
+season.  Afterward, in the `diff`, see how we changed some of the colors:
+
+```diff
+diff --git a/style/variables.css b/style/variables.css
+index f8c738f..9f39f14 100644
+--- a/style/variables.css
++++ b/style/variables.css
+@@ -87,9 +87,9 @@ all of MD as it is not optimized for dense, information rich UIs.
+    */
+
+   --jp-border-width: 1px;
+-  --jp-border-color0: var(--md-grey-400);
+-  --jp-border-color1: var(--md-grey-400);
+-  --jp-border-color2: var(--md-grey-300);
++  --jp-border-color0: #168EA8;
++  --jp-border-color1: #168EA8;
++  --jp-border-color2: #93D3E1;
+   --jp-border-color3: var(--md-grey-200);
+   --jp-border-radius: 2px;
+
+@@ -118,10 +118,10 @@ all of MD as it is not optimized for dense, information rich UIs.
+```
+
+I'd like the "running notebook" dot in the filesystem browser to be a
+snowman instead of a blue dot. Using the browser inspector, I can look a the CSS
+doing this and override it in my theme:
+
+[Is the dot in the second line on the right side of this code block intentional?]
+```diff
+@@ -368,0 +369,26 @@                                                                 .
++
++.jp-DirListing-item.jp-mod-running .jp-DirListing-itemIcon:before {
++    content: '\2603'; /* snowman */
++    font-size: 10px;
++    position: absolute;
++    left: -8px;
++}
+```
+
+We can also add backgrounds to many of the panels.  By adding a file `snowflakepatterns.svg` to our style directory,
+we can now refer to it from our file and add the following CSS to include snow flakes in the background of 
+our directory listing.
+
+```css
+/* DirListing */
+
+body[data-jp-theme-name="JupyterLab Winter"] .jp-DirListing-content {
+  background-image: url('./snowflakepattern.svg');
+ }
+```
+
+# Awesome!
+
+Here is the final result, jupyterLab-theme-winter theme provided by QuanSight.  Feel
```

Comment:
Still not sure if you actually are going to link a repo, but I suspect yes and then `jupyterLab-theme-winter` should be typeset as code.

Reply:
the images are links, I also linked the `Here`.